server/views.py

```diff
@@ -182,6 +183,95 @@ def card(request, tool_id, card_id):
 
   return makeResponse(request, result)
 
+@check_secret
+@check_ip
+@require_GET
+def getinfo(request, tool_id, item_requested):
+  ip = get_ip(request)
+  try:
+    t = Tool.objects.get(pk=tool_id)
+  except ObjectDoesNotExist as e:
+    logger.warning('Tool does not exist for %s // %s from %s', request.method, request.path, ip,
+                extra={
+                    'status_code': 200,
+                    'request': request
+                }
+            )
+    result = { 'numeric_status' : -1, 'error' : 'Tool does Not Exist' }
+    return makeResponse(request, result)
+
+  try:
+    i = Venditem.objects.get(item=item_requested)
+  except ObjectDoesNotExist as e:
+    logger.warning('Item does not exist %s // %s from %s', request.method, request.path, ip,
+                extra={
+                    'status_code': 200,
+                    'request': request
+                }
+            )
+    result = { 'numeric_status' : -1, 'error' : 'Item does Not Exist' }
+    return makeResponse(request, result)
+
+  result = { 'numeric_status' : 1 }
+  # For subscribed users, getting the name is useful for doorbot
+  result['item_name'] = i.name
+  result['item'] = i.item
+  result['price'] = i.price
+  result['stock'] = i.stock
+
+  return makeResponse(request, result)
+
+
+@check_secret
+@check_ip
+@require_GET
+def updatebalance(request, tool_id, new_balance, card_id):
+  try:
+    t = Tool.objects.get(pk=tool_id)
+  except ObjectDoesNotExist as e:
+    result = { 'numeric_status' : -1, 'error' : 'Tool does not exist'}
+    return makeResponse(request, result)
+
+  try:
+    c = Card.objects.get(card_id=card_id)
+  except ObjectDoesNotExist as e:
+    result = { 'numeric_status' : 0, 'error' : 'Card does not exist'}
+    return makeResponse(request, result)
+
+  if not c.user.subscribed:
+    # needs to be subscribed to be a user
+    result = { 'numeric_status' : 0, 'error' : 'Card is not subscribed'}
+    return makeResponse(request, result)
+
+  c.user.balance = new_balance
+  c.user.save()
```

Comment:
How do you deal with failures here? do you return the coin? if not, we need some logging of failures - perhaps the machine needs an endpoint it can log its own status to for this sort of thing. Perhaps we should make use of syslog?